### ISSUE TITLE
steam UI uses windows-style vertical scroll bar

### ISSUE THREAD

**Issue by craig-sanders** [ValveSoftware/steam-for-linux#4231]
The Steam UI does the same irritating thing that Windows scroll bars do - if you are scrolling up or down and the mouse pointer moves too far (only an inch or so) to the LEFT or RIGHT, the scrolled page snaps back to its original position.

This is supremely irritating as you have to be extremely careful not to move the mouse horizontally at all.

Scroll bars do not work like this on linux, AFAIK there isn't even one window manager or desktop environment which does this (and certainly not by default) - while scrolling vertically you can move the mouse horizontally as much as you like and it makes no difference at all.

I avoid horizontal scroll bars by resizing windows so i don't know if a similar thing happens for horiz bars when moving the mouse "too far" up or down.

please fix scrollbars make them work like other linux apps....or at least make the "too far" setting configurable so i can set it to something like 10000px on either side.

**Comment by kisak-valve** [comment]
Hello @craig-sanders, are you still experiencing this issue on an up to date system?

**Comment by craig-sanders** [comment]
No, the behaviour has changed.  Now:

1.  The scroll bar completely ignores vertical mouse movement unless the mouse cursor is directly over the scrollbar.  This is not good, but it's far less annoying than immediately snapping back to where scrolling started when the mouse moves "too far" to either side.

2. The scrollbar has a strong tendency to ignore the fact that the mouse button has been released, causing the scroll bar to still scroll up and down.  This seems to happen most often when the page (e.g. the store page for a game) is loading more details (such as user reviews).   To get it to stop scrolling when this happens, you have to click and release on the scrollbar again.   This is not due to my system being under-powered (it's a Threadripper 1950x with 64GB RAM and an nvidia GTX-970 using the proprietary nvidia driver, with a pair of mirrored Samsung 970 NVME drives and a reasonably fast ADSL2 internet connection - i.e. nothing that would cause any lags or delays).

In short, scroll bar behaviour is still buggy and disconcertingly weird, but in a different manner than before.

**Comment by kisak-valve** [comment]
What DE are you using?

**Comment by craig-sanders** [comment]
xfce 4.12

**Comment by Chiitoo** [comment]
Does it work as expected in the Library view for you?  For me it does.

However, in the Inventory and Store views for example (anything that uses the browser I imagine), as long as my cursor stays within the Steam window, everything's fine, but if I go outside it at all during scrolling (that is, while having clicked and holding down the left mouse button on the bar to 'grab' it), the bar is 'stuck' in scrolling mode.  That is, if I move the mouse outside while 'holding on' to the scroll bar, no matter if I release the left button or not, the bar will stop moving, until I move the cursor back into the window (however, the bar will only move while the cursor is within the scroll bar area then).

Clicking on the bar will 'reset' things back to normal.

This is using KWin as the window manager.

Been meaning to report this as well, as it is quite annoying indeed.  :]

**Comment by kisak-valve** [comment]
1 looks a DE-specific issue I'm able to grab the scroll bar and move the mouse any distance away from it in a quick test with KDE and MATE

2 looks like a separate libcef issue, which you can check by comparing the scroll bar in the store tab versus the library tab (the latter should not be affected). Please open a separate issue report for this.

**Comment by craig-sanders** [comment]
> Does it work as expected in the Library view for you? For me it does.

Yes, Library view works fine.  no problem there.

> However, in the Inventory and Store views for example (anything that uses the browser I imagine), as
> long as my cursor stays within the Steam window, everything's fine, but if I go outside it at all during
> scrolling

the problem occurs for me whether the cursor stays within the steam window or not.

**Comment by Chiitoo** [comment]
Actually, it seems the cursor doesn't need to leave the window altogether, but simply what I guess is the 'cef' area (so moving onto the gray/green'ish Steam window borders is enough).

Tested now with kwin_x11, openbox, and xfwm4 on LXQt, and all seem to behave similarly to me.  I have not managed to see it yet by moving to the left (unless I go far enough to hit the borders, be it up, down, or left).

**Comment by kisak-valve** [comment]
@craig-sanders, can you check XFCE Settings -> Window Manager -> Focus mode and let me know if Focus Follows Mouse is selected? There are long standing issues with that setting going back to #6.

**Comment by craig-sanders** [comment]
> Focus Follows Mouse 

I see you've already struck this out and said it's unrelated, but for completeness: No, i use click-to-focus.  I can't stand focus follows mouse.

**Comment by Chiitoo** [comment]
It might be at least a tad obvious, but this now affects the library view as well.

**Comment by kokoko3k** [comment]
It affects me since ages and i tought it was just me.
I'm still affected, using kde.
